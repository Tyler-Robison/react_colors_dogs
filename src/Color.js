import { useParams, useNavigate } from 'react-router-dom';
import { useEffect } from 'react';

const Color = ({ colors }) => {

    const navigate = useNavigate();
    const { color } = useParams()

    // re-direct to color select if color isn't valid
    useEffect(() => {
        if (!colors.includes(color)) navigate('/')
    }, [])

    // console.log('color', color)
    // console.log('colors', colors)

    return (
        <div data-testid={`${color}-div`}  style={{ backgroundColor: `${color}`, height: '80vh', width: '100vh' }}>

        </div>
    )
}

export default Color;