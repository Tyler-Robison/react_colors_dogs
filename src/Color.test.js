import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Color from "./Color";
import { MemoryRouter } from "react-router-dom";

const colors = ['red', 'blue', 'green'];

it("renders without crashing", function () {
    <MemoryRouter initialEntries={['/colors/red']}>
         <Color colors={colors} />
    </MemoryRouter>
});

it("matches snapshot", function () {
    const container = render(
        <MemoryRouter initialEntries={['/colors/red']}>
          <Color colors={colors} />
        </MemoryRouter>
      );
      expect(container.asFragment()).toMatchSnapshot();
});

