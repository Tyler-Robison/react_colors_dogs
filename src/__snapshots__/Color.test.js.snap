// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="undefined-div"
    style="height: 80vh; width: 100vh;"
  />
</DocumentFragment>
`;
